src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -53,44 +53,45 @@ public final class InlineConfigParser {
     /**
      * A pattern that matches the following comments formats.
      * <ol>
-     *     <li> "// violation" </li>
-     *     <li> "// violation, 'violation message'" </li>
-     *     <li> "// violation 'violation messages'" </li>
+     *     <li> // violation </li>
+     *     <li> // violation, 'violation message' </li>
+     *     <li> // violation 'violation messages' </li>
+     *     <li> // violation, "violation messages" </li>
      * </ol>
      *
      * <p>
      * This pattern will not match the following formats.
      * <ol>
-     *     <li> "// violation, explanation" </li>
-     *     <li> "// violation, explanation, 'violation message'" </li>
+     *     <li> // violation, explanation </li>
+     *     <li> // violation, explanation, 'violation message' </li>
      * </ol>
      *
      * These are matched by
      * {@link InlineConfigParser#VIOLATION_WITH_EXPLANATION_PATTERN}.
      * </p>
      */
     private static final Pattern VIOLATION_PATTERN = Pattern
-            .compile(".*//\\s*violation,?\\s*(?:'(.*)')?$");
+            .compile(".*//\\s*violation,?\\s*(?:['\"](.*)['\"])?$");
```

Comment:
We have a lot of check messages with parameters that are enclosed with `'`. We should enclose the check message in the BDD comment with `"` in this case, we have no reason to be restrictive.

Reply:
Why be so strict? If we aren’t having problems parsing, I don’t see why we should care.

Reply:
the reason I persuaded myself is we can use `"message 'token'"` instead of ``'message \'token\'``

Reply:
>The same reason we are strict on a lot of other thing, just for uniform conformity.

This is natural language we are dealing with in this case, if it sounds/reads well (as in @romani 's example right above this), it is good.

Reply:
ok.